src/appleseed.studio/mainwindow/project/attributeeditor.cpp

```diff
@@ -83,9 +85,12 @@ void AttributeEditor::edit(
         m_entity_editor.get(), SIGNAL(signal_applied(foundation::Dictionary)),
         receiver, slot_apply);
 
-    QObject::connect(
-        m_parent->window(), SIGNAL(signal_crop_window_cleared()),
-        m_entity_editor.get(), SLOT(slot_clear_crop_window_widget()));
+    FrameItem* frame_item = dynamic_cast<FrameItem*>(receiver);
```

Comment:
This is the problematic part of this PR (and of the previous one, a fact that I missed in my first review):

The attribute editor should not have logic for specific types of items. It's a generic facility that should work in a generic manner.